### DISCUSSION TITLE
Can (deprecated) Native Client concept and .nexe file be useful here?

### DISCUSSION THREAD

**Discussion by guest271314** [nodejs/single-executable#12] [Ideas]
TL;DR (https://github.com/WebAssembly/wabt/issues/1864; https://github.com/GoogleChrome/developer.chrome.com/issues/2295; https://stackoverflow.com/q/70694704) I unsuccessfully tried to reverse-engineer tts_service_x86_64.nexe, a Google speech synthesis engine executable used for [US English Female Text-to-speech (by Google)](https://chrome.google.com/webstore/detail/us-english-female-text-to/pkidpnnapnfgjhfhkpmjpbckkbaodldb).

The `strings` the executable of tts_service_x86_63.nexe https://gist.github.com/guest271314/4894662d31ffb4ad2cd57683721bf4b4.

[NaCl and PNaCl](https://developer.chrome.com/docs/native-client/nacl-and-pnacl/)
> Portable Native Client (PNaCl)
> PNaCl solves the portability problem by splitting the compilation process into two parts:
> 
> - compiling the source code to a bitcode executable (pexe), and
> - translating the bitcode to a host-specific executable as soon as the module loads in the browser but before any code execution.
> ...
> With PNaCl, you’ll generate a single pexe, rather than multiple platform- specific nexes...

From [Aligned bundling support in LLVM](https://www.chromium.org/nativeclient/pnacl/aligned-bundling-support-in-llvm/) at [Implementation in LLVM MC](https://www.chromium.org/nativeclient/pnacl/aligned-bundling-support-in-llvm/#implementation-in-llvm-mc)

> The following description will focus on the path: Text assembly -> ELF object streamer -> Assembly -> Object file emission. This path can be roughly divided to three stages:
> 
> 1. Parse bundling directives and emit a sequence of sections (MCSectionData) and fragments (MCFragment derivatives) that represent the input.
> 2. Perform layout of the fragments to be valid w.r.t. relaxation and bundling.
> 3. Write the output object.

**Comment by RaisinTen** [comment]
I didn't dig too deep but is this about creating a single binary that would run on all platforms?
